internal/parse-options.js

```diff
@@ -1,11 +1,46 @@
-// parse out just the options we care about so we always get a consistent
-// obj with keys in a consistent order.
-const opts = ['includePrerelease', 'loose', 'rtl']
-const parseOptions = options =>
-  !options ? {}
-  : typeof options !== 'object' ? { loose: true }
-  : opts.filter(k => options[k]).reduce((o, k) => {
-    o[k] = true
-    return o
-  }, {})
+// parse out just the options we care about
+const isParsedConfigSymbol = Symbol('isParsedConfig');
+const var1 = Object.freeze({ includePrerelease: true, loose: true, rtl: true, [isParsedConfigSymbol]: true })
+const var2 = Object.freeze({ includePrerelease: true, loose: true, [isParsedConfigSymbol]: true })
+const var3 = Object.freeze({ includePrerelease: true, rtl: true, [isParsedConfigSymbol]: true })
+const var4 = Object.freeze({ includePrerelease: true, [isParsedConfigSymbol]: true })
+const var5 = Object.freeze({ loose: true, rtl: true, [isParsedConfigSymbol]: true })
+const var6 = Object.freeze({ loose: true, [isParsedConfigSymbol]: true })
+const var7 = Object.freeze({ rtl: true, [isParsedConfigSymbol]: true })
```

Comment:
might be nice to have better variable names here :-)

Reply:
Agreed.  Starting from the perspective of a general roadmap of `make it, make it right, make it fast` this breaks `make it right` in that it is much much harder for a developer to follow.  I think in the balance between speed and developer experience this goes just a shade too far.